daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -105,8 +105,14 @@ object VariableUtils {
     bindings.foreach { b => currentVMap.setExtVariable(b.varQName, b.varValue, referringContext) }
   }
 
-  def convert(v: String, rd: VariableRuntimeData): DataValuePrimitive =
-    rd.primType.fromXMLString(v)
+  def convert(v: String, rd: VariableRuntimeData, referringContext: ThrowsSDE): DataValuePrimitive = {
+    try {
+      rd.primType.fromXMLString(v)
+    } catch {
+      case e: Throwable => referringContext.SDE("Error processing variable %s: %s".format(rd.globalQName, e.getMessage))
```

Comment:
Don't catch throwable here. Use narrower set of exceptions, which you should now have a good handle on.